### ISSUE TITLE
Crash loading STL file

### ISSUE THREAD

**Issue by Timmmm** [ElmerCSC/elmerfem#566]
Using the Windows binary download from Sourceforge, compiled on Feb 7 2024, I opened this STL file:

[bar.zip](https://github.com/user-attachments/files/16936512/bar.zip)

Elmer then crashes.

By the way the UX of loading meshes is not great:

1. There's File->Open and File->Load Mesh. Apparently to load a mesh you use File->Open, not File->Load Mesh?? And for some reason File->Open opens a file and File->Load Mesh opens a directory... Very confusing.
2. Neither option lists the file formats that it supports. Fair enough for File->Load Mesh, but File->Open definitely should!
3. I initially tried to load `bar.STL` (Solidworks' default extension), and *very luckily* I noticed the message in the status bar saying "make sure the file extension is lowercase", but I very easily could have missed that, and why do I even need to do that anyway? Just lowercase it yourself if you need to! At the very least you can detect if it isn't lowercase and throw up a modal error.

**Comment by Timmmm** [comment]
Possibly the same as #182 though that says it has been fixed...

**Comment by mmuetzel** [comment]
Thank you for your report.

Do you still see this issue when you test with a nightly build that you could download from https://www.nic.funet.fi/pub/sci/physics/elmer/bin/windows/ ?

**Comment by Timmmm** [comment]
Yep.

**Comment by richb2k** [comment]
PR #618  addresses most of the above issues.  ToLower was added, so the file suffix can be either upper or lower case.  

The binary stl file from Issue #566, bar.stl, now loads and is read, such as reading the proper number of facets, but then ElmerGui crashes during the meshing operation.  The binary stl file from Issue #182, 02-cores.stl, loads and is read, and then is successfully meshed and displayed.  

In other words, both binary files now load okay, but nglib has a problem meshing bar.stl and has no problem meshing 02-core.stl.
